core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -82,7 +82,11 @@ public void configurePipeline(MultiInputPipelineConfigurer pipelineConfigurer) {
     init(inputSchemas, collector);
     collector.getOrThrowException();
     //validate the input schema and get the output schema for it
-    stageConfigurer.setOutputSchema(getOutputSchema(inputSchemas, collector));
+    Schema outputSchema = getOutputSchema(inputSchemas, collector);
```

Comment:
does getOutputSchema handle null join keys and other configs?

Reply:
Yes.
null join keys (configured by macro) is handled by JoinerConfig.validateJoinKeySchemas().
JoinerConfig.getInputs() returns empty set when requiredInputs is null.
getOutputSchema returns null if selectedFields is null.